### PR TITLE
Adds new space ruin: Luxury Vehicle Transport

### PR DESCRIPTION
**Pull request by Robustin**
Newcar

Space could use some more variety so I added a short and simple ruin, image attached. Speederbikes can be retrieved for 1,000 credits, Speedwagon is 4,000 credits (speederbike with an impact effect).

A hardlight blade is the only other notable item, its basically a bulky esword that cannot be bagged/pocketed/turned off. 

There are two teleportation turrets to serve as a defensive mechanic. If anyone has design suggestions I'd be happy to hear it.

### CODE REVIEW

**Comment by calzilla1** [comment]
I like the GPS

**Comment by Niknakflak** [comment]
That's a pretty ugly shuttle, not opposed to the idea but dang, it's a rectangle with a three engine strapped to the back

**Comment by optimumtact** [comment]
it's a space ruin niknak

**Comment by Cheridan** [comment]
it's really very ugly

**Comment by kevinz000** [comment]
1. Why are you using something that is used in cybernetic implants as a standalone item I don't get..
2. Please take a very, very close look at https://github.com/tgstation/tgstation/blob/be7d838789d58d8e30156cb8aad78484e19b9eb4/code/modules/vehicles/speedbike.dm#L49 before you consider putting in a speedwagon.

**Comment by PKPenguin321** [comment]
yeah, fine idea but the car is 1. planned to be a wizard item afaik (it's really strong) 2. i think it's using an energy sword subtype that isnt supposed to exist on its own 3. it looks really gross and rectangular

**Comment by KorPhaeron** [comment]
>adding an empty loot box with no challenge so you can go grab admin and wizard items as assistant 

Hmm

**Comment by Armhulen** [comment]
We need more movable space ships.

**Comment by kevinz000** [comment]
@Armhulen if only shuttle code allowed for user constructed shuttles.

File: code/datums/ruins/space.dm [hunk 1]
```diff
@@ -242,6 +242,12 @@
 	suffix = "miracle.dmm"
 	name = "Ordinary Space Tile"
 	description = "Absolutely nothing strange going on here please move along, plenty more space to see right this way!"
+	
+/datum/map_template/ruin/space/luxuryvehicle
+	id = "luxuryvehicle"
+	suffix = ".dmm"
```
Comment (kevinz000) [inline code/datums/ruins/space.dm]: you forgot the filename

File: code/datums/ruins/space.dm [hunk 2]
```diff
@@ -242,6 +242,12 @@
...
+	name = "A luxury vehicle showcase ambushed by pirates, hope you brought cash"
```
Comment (kevinz000) [inline code/datums/ruins/space.dm]: this should be the desc

File: code/datums/ruins/space.dm [hunk 3]
```diff
@@ -242,6 +242,12 @@
...
+	description = "Absolutely nothing strange going on here please move along, plenty more space to see right this way!"
```
Comment (kevinz000) [inline code/datums/ruins/space.dm]: bad

**Comment by Robustin** [comment]
If to my absolute astonishment these vehicles end up influencing too many rounds negatively ill be the first to up the credit cost. 

I'm surprised you'd complain about this Kor since half of my mining shifts have a nukeop hardsuit or gluttony syringe sitting at my doorstep "for no challenge". The challenge for this is acquiring the necessary credits. I'd contemplated several puzzles/enemies but they all just ended up looking like the Caravan Ambush, hopeless for ordinary spess explorers but an easy treasure room for people who know how to exploit simple mob AI.

**Comment by Robustin** [comment]
I've doubled the cost of both vehicles (4000/1000) and added a bluespace turret (think wand of teleportation, as a turret) for an added layer of challenge - to address concerns.

**Comment by Cheridan** [comment]
Can I not just weld/blow the walls down and laugh

**Comment by ExcessiveUseOfCobblestone** [comment]
Don't worry this can't be abused because no one goes into space!

**Comment by MMMiracles** [comment]
you used indestructible doors for the irrelevant airlocks but used regular walls to protect the actual loot.

robustin i dont think you thought this through very well

**Comment by Robustin** [comment]
I'm uploading a corrected version.

**Comment by Tofa01** [comment]
Just a note but i think you put the thrusters facing the wrong direction.

**Comment by Lexorion** [comment]
The hardlight blade is supposed to be an implant, hence you can't turn it off (just toggle it,normally). It'd be better if you replace the weapon with the actual implant instead. That aside, I like the idea of being able to purchase gear in a ruin,maybe we'll have an actual merchant shuttle (not the admin event) one day.

**Comment by KorPhaeron** [comment]
@Robustin 

A hardsuit on lavaland (a place that can and will kill people regularly) is not the same as an unlimited max speed, stun on contact weapon

**Comment by ExcessiveUseOfCobblestone** [comment]
>  That aside, I like the idea of being able to purchase gear in a ruin,maybe we'll have an actual merchant shuttle (not the admin event) one day.

I wonder if you could make special display cases that randomize items associated with a set cost [IE it rolls between X for 500, Y for 1000, Z for 100000, etc.] and can't break. Then you can just plop those in this shuttle and voila!

**Comment by Robustin** [comment]
If the car is a wizard meme I can see why it would be strong in combination with all the other spells they get.

But for a standalone space explorer, even if they were an antag, all it takes is one stun and they're done. 

If the car's strength is really the thing holding this ruin back I will go in and code a child that requires a car key so you can't have double esword memes on it. 

Speaking personally I wouldnt even be mad if someone killed me with the car. That shit sounds hilarious.

**Comment by Robustin** [comment]
@Tofa01 

Thats just how Dream Maker displays things. The engines are set to dir = 6, which means they're correctly aligned.

**Comment by Jordie0608** [comment]
You sure about that? Dir 6 is southeast and afaik engines don't have diagonals.

**Comment by optimumtact** [comment]
is it meant to look like a giant gold phallus?

**Comment by WJohn** [comment]
come on man, you know better than to have diagonally unconnected walls. you've seen this for years, you should know not to do that.

**Comment by Robustin** [comment]
Plz tell me how to diagonal, halp computer.

Also plz stop make fun of shape, u now have 1 hour to name a shuttle that is not box or phallus, go.

**Comment by KorPhaeron** [comment]
Ugly and pointless

Some lavaland ruins are also ugly and pointless but that doesnt mean we need more like them

**Comment by KorPhaeron** [comment]
I guess the trader concept is kind of clever though maybe its salvageable

**Comment by Robustin** [comment]
I will give muh aesthetics one more pass once I figure out how to smooth the corners.

Other than that it gives the crew something do in longer rounds that isn't stupidly difficult (Paper Dungeon) or just a giant syndicate giveaway (Caravan Ambush), and not just a floating abandoned room (everything else).

**Comment by WJohn** [comment]
Syndicate walls do not have automatic smoothing. Instead, they use unchanging diagonal blocks to do it, both interior and exterior versions. Only white shuttle walls support smoothing (and diagonal smoothing at that).

Regardless, you should never leave diagonal unconnected walls like that in any situation. I know mmmiracles is notorious for doing it, and it's really ugly. Please avoid doing it.

**Comment by Robustin** [comment]
Will the walls appear as "corner" if I set their dir to a diagonal one? 

Not being able to import these ruins for local testing is making this really hard.

**Comment by Tofa01** [comment]
## **How to force spawn a ruin at least my way of doing it**

Step 1 is to define the ruins .dmm path in the include of whatever map you are compiling.

Step 2 is to make sure that that map is ticked in DM to allow it to compile (Not sure if this is necessary but i do it anyway.


Step 3 compile.

Step 4 go ingame as a observer and you should now see your ruin under the teleport options.

**Comment by XDTM** [comment]
There's a command to just spawn a ruin, although it may or may not be broken right now.

**Comment by Tofa01** [comment]
Last time I tried it, it did not have all ruins in the list.

On 2 Mar 2017 8:31 am, "XDTM" <notifications@github.com> wrote:

There's a command to just spawn a ruin, although it may or may not be
broken right now.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/tgstation/tgstation/pull/24449#issuecomment-283476793>,
or mute the thread

**Comment by Jordie0608** [comment]
@Tofa01 Only one of the first two steps are needed you don't need to tick a .dmm file if you `#include` it, and vice versa.

**Comment by Robustin** [comment]
Ok I'm doing the visuals pass today, last call for any specific requests.

**Comment by calzilla1** [comment]
A few abandoned crates would be good clutter

**Comment by Robustin** [comment]
I definitely add some aux loot, though abandoned crates were bugged last I checked.

I'm trying to get my engi PR merged so I can swap out the vehicles too.

**Comment by optimumtact** [comment]
ping for reopen when you finally finish your engie pr or whatever

**Pull request closed without merging**